File: samples-java/src/main/java/io/github/ilaborie/loops/samples/anatomy/Nest.java
package io.github.ilaborie.loops.samples.anatomy;

import io.github.ilaborie.loops.samples.Element;

import java.util.ArrayList;
import java.util.List;
import java.util.Objects;

import static io.github.ilaborie.loops.samples.Elements.getSomeElements;

public class Nest {

    public static void main(String[] args) {
        List<Element> elements = getSomeElements();
        List<String> result = nest(elements);
        result.forEach(System.out::println);
    }

    public static List<String> nest(List<Element> elements) {
        List<String> result = new ArrayList<>();

        for (Element elt1 : elements) {
            for (Element elt2 : elements) {
                String name1 = elt1.getName();
                String name2 = elt2.getName();
                if (!Objects.equals(name1, name2) &&
                        name1.length() == name2.length()) {
                    result.add(String.join("-", name1, name2));
                }
            }
        }

        return result;
    }

}

File: samples-kotlin/src/main/kotlin/io/github/ilaborie/loops/samples/anatomy/nest.kt
package io.github.ilaborie.loops.samples.anatomy

import io.github.ilaborie.loops.samples.ElementKt
import io.github.ilaborie.loops.samples.getSomeElementsKt


fun main() {
    val elements = getSomeElementsKt()
    val result = io.github.ilaborie.loops.samples.stream.nest(elements)
    result.forEach { println(it) }
}

fun nest(elements: List<ElementKt>): List<String> {
    val result = mutableListOf<String>()

    for (elt1 in elements) {
        for (elt2 in elements) {
            val name1 = elt1.name
            val name2 = elt2.name
            if (name1 != name2 && name1.length == name2.length) {
                result.add(arrayOf(name1, name2).joinToString("-"))
            }
        }
    }

    return result
}